File: Objective-C/Reference App/12. Create Map View Controller/CreateMapVCViewController.m
//
//  CreateMapVCViewController.m
//  Reference App
//
//  Created by David Deller on 5/3/18.
//  Copyright © 2018 TallyGo. All rights reserved.
//

#import "CreateMapVCViewController.h"

@interface CreateMapVCViewController ()

@end

@implementation CreateMapVCViewController

- (IBAction)go:(id)sender {
    [self showMapVC];
}

- (UIViewController *)menuViewController {
    UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"This is where your custom menu would go." message:nil preferredStyle:UIAlertControllerStyleAlert];
    [alert addAction:[UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:nil]];
    [alert addAction:[UIAlertAction actionWithTitle:@"Exit" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
        [self dismissViewControllerAnimated:YES completion:nil];
    }]];
    
    return alert;
}

- (void)showMapVC {
    TGMapViewController *mapViewController = [TGMapViewController create];
    __block TGMapViewController *blockVCRef = mapViewController;
    mapViewController.onSelectMenuButton = ^{
        [blockVCRef presentViewController:self.menuViewController animated:YES completion:nil];
    };
    
    [self presentViewController:mapViewController animated:YES completion:nil];
}

@end

File: Swift 4/Reference App/201. Map, search, and navigation with TGMapViewController/CreateMapVCViewController.swift
//
//  CreateMapVCViewController.swift
//  Reference App
//
//  Created by David Deller on 5/1/18.
//  Copyright © 2018 TallyGo. All rights reserved.
//

import UIKit
import TallyGoKit

class CreateMapVCViewController: ExampleViewController {

    @IBAction func go(_ sender: Any) {
        showMapVC()
    }
    
    // MARK: -
    
    var menuViewController: UIViewController {
        let alert = UIAlertController(title: "This is where your custom menu would go.", message: nil, preferredStyle: .alert)
        alert.addAction(UIAlertAction(title: "Cancel", style: .default, handler: nil))
        alert.addAction(UIAlertAction(title: "Exit", style: .default, handler: { (action) in
            self.dismiss(animated: true, completion: nil)
        }))
        
        return alert
    }
    
    func showMapVC() {
        let mapViewController = TGMapViewController.create()
        mapViewController.onSelectMenuButton = {
            mapViewController.present(self.menuViewController, animated: true)
        }
        
        present(mapViewController.embedInNavigationController(), animated: true)
    }

}
